missed-moment.py

```diff
@@ -133,26 +162,31 @@ def start_audio_server(device_id):
     # -d: driver backend
     #   -dalsa -C provide only capture ports
     #   -p: period - This value must be a power of 2, and the default is 1024
+    #       If the period should be set as low as possible to not get these types of errors.  Start with 128, go up through 256, 512, 1024, 2048 and so on.
+    # raspberrypi python3[6384]: JackEngine::XRun: client = jack_capture was not finished, state = Triggered
+    # raspberrypi python3[6384]: JackAudioDriver::ProcessGraphAsyncMaster: Process error
+    # raspberrypi python3[6384]: JackEngine::XRun: client = jack_capture was not finished, state = Running
+    # raspberrypi python3[6384]: JackAudioDriver::ProcessGraphAsyncMaster: Process error
     #   -n: number of periods of playback latency
     #   -r: sample rate, default is 48000
     #   -s: softmode - this makes jack less likely to disconnect unresponsive ports
     #       when running without --realtime
     logging.info("Starting audio server")
-    result = Popen("ps -ef | grep [j]ackd | awk '{print $2}'", shell=True, stdout=subprocess.PIPE)
-    audio_server_pid_string = result.stdout.read().decode('utf-8').split('\n')
-    for line in audio_server_pid_string:
-        if line:
-            logging.debug(f'stopping already running jackd {line}')
-            kill(int(line), SIGTERM)
+    stop_audio_server()
```

Comment:
I'm not sure if I should take this out or not from here.  I moved the finally block out to encompass all code, which calls this.  So I shouldn't have to have this here.  It's a precaution to stop an already running server prior to starting it again. Thoughts?